js/main.js

```diff
@@ -25,36 +29,63 @@ function submitForm(event) {
   var titleValue = newEntryForm.elements.title.value;
   var urlValue = newEntryForm.elements['photo-url'].value;
   var notesvalue = newEntryForm.elements.notes.value;
-  inputValues.title = titleValue;
-  inputValues.url = urlValue;
-  inputValues.notes = notesvalue;
-  inputValues.entryId = data.nextEntryId;
-  data.entries.unshift(inputValues);
-  data.nextEntryId++;
-  newEntryForm.reset();
-  var newEntry = renderEntry(data.entries[0]);
-  ulElement.prepend(newEntry);
+  if (data.editing !== null) {
+    var entry = document.querySelector('[data-entry-id=' + CSS.escape(data.editing.entryId) + ']');
```

Comment:
`CSS.escape` shouldn't be necessary here, you can just concatenate the value of the property directly.